virtool/indexes/utils.py

```diff
@@ -12,3 +21,22 @@ def check_index_file_type(file_name: str) -> str:
         return "json"
     else:
         return "bowtie2"
+
+
+async def check_file_exists(pg: AsyncEngine, filename: str, index: str) -> Optional[Base]:
+    """
+    Check if a index file already exists in `IndexFile` SQL table.
+
+    :param pg: the PostgreSQL AsyncEngine object
+    :param filename: the name of the index file
+    :param index: the id of the index
+
+    :return: `None` or a row from `IndexFile` table
+
+    """
+    async with AsyncSession(pg) as session:
+        exists = (await session.execute(
+            select(IndexFile).filter_by(name=filename, index=index)
+        )).scalar()
```

Comment:
Is there more efficient way than fetching entire row? It is not super important here, but I would like to have an example of how to do this.

Reply:
I haven't found a better solution for this.
There is an [exists()](https://docs.sqlalchemy.org/en/14/orm/query.html?highlight=exists#sqlalchemy.orm.Query.exists) method in SQLAlchemy, but can't work with `AsyncSession`

Reply:
Okay. Thanks for checking.